script/source-setup/npm

```diff
@@ -15,20 +15,14 @@ if [ "$1" == "-f" ]; then
   FORCE=1
 fi
 
-NPM_MAJOR_VERSION="$(npm -v | cut -d'.' -f1)"
-if [ "$NPM_MAJOR_VERSION" -ge "7" ]; then
-  PACKAGE_JSON_SRC="package.json.npm7"
-else
-  PACKAGE_JSON_SRC="package.json.npm6"
-fi
-
-if [ ! -f "package.json" ] || [ "$(cat package.json | md5sum )" != "$(cat "$PACKAGE_JSON_SRC" | md5sum)" ]; then
-  FORCE=1
-  cp -f "$PACKAGE_JSON_SRC" package.json
-fi
-
 if [ -n "$FORCE" ]; then
   find . -not -regex "\.*" -and -not -name "package\.json*" -print0 | xargs -0 rm -rf
 fi
 
-npm install
+NPM_MAJOR_VERSION="$(npm -v | cut -d'.' -f1)"
+if [ "$NPM_MAJOR_VERSION" -ge "7" ]; then
+  # do no install peerDependencies in npm 7
+  npm install --legacy-peer-deps
+else
+  npm install
+fi
```

Comment:
I was curious about what worked or didn't work when installing with the flag or not and checked out the branch locally to play around with things.  It looks like all the tests pass as expected whether the flag is set or not 🤷 .  Do you know of any reason why installing with the flag is needed or can it be reverted?

Reply:
huh, when I installed locally with npm7 (fresh setup) without the flag it also installed the `@react-native-community/async-storage` dependency, which failed some tests
I'm going to try again and see what happens

Reply:
huh I must have put things into a weird state locally 🤷 .  shouldn't block this PR, happy to leave this as-is and I can figure it out later